src/mon/MonCap.h

```diff
@@ -196,13 +196,22 @@ struct MonCap {
   }
 
   bool fs_name_capable(string_view fs_name, __u8 mask) {
-    for (auto& g: grants) {
+    for (auto& g : grants) {
       if (g.is_allow_all()) {
 	return true;
       }
-      if (g.fs_name.empty() || g.fs_name == fs_name) {
-	if (mask & g.allow) {
+
+      if ((g.fs_name.empty() || g.fs_name == fs_name) && (mask & g.allow)) {
 	  return true;
+      }
+
+      if (not g.profile.empty()) {
+	for (auto& pg : g.profile_grants) {
```

Comment:
profile_grants is populated by MonCapGrant::expand_profile, which is called by MonCap::is_capable. Are we always sure that  MonCap::is_capable will be called before MonCap::fs_capable?

Reply:
IMO, the cleaner way to fix this is to modify "MonCapGrant::get_allowed" to honor fs_name caps and validate it.

Reply:
Fixed, thanks!